### PR TITLE
ec2: Update documentation

### PR DESCRIPTION
**Pull request by alinabuzachis**
##### SUMMARY

Document returned values.
Fixes: https://github.com/ansible-collections/amazon.aws/issues/308

##### ISSUE TYPE

- Docs Pull Request

##### COMPONENT NAME

ec2

### CODE REVIEW

File: plugins/modules/ec2.py [hunk 1]
```diff
@@ -577,6 +577,297 @@
...
+            returned: always
+            sample: {
+                "/dev/xvda": {
+                    "delete_on_termination": true,
+                    "status": "attached",
+                    "volume_id": "vol-06d364586f5550b62"
+                }
+            }
+        dns_name:
+            description: If the EC2 instance has changed.
```
Comment (jillr) [inline plugins/modules/ec2.py:613]:
```suggestion
            description: The public DNS name assigned to the instance.
```
I'm reasonably certain it's the public address at least, http://boto.cloudhackers.com/en/latest/ref/ec2.html#boto.ec2.instance.Instance doesn't document dns_name.

File: plugins/modules/ec2.py [hunk 2]
```diff
@@ -577,6 +577,297 @@
...
+            type: str
+            returned: always
+            sample: ""
```
Comment (jillr) [inline plugins/modules/ec2.py:616]:
```suggestion
            sample: "ec2-203-0-113-1.z-2.compute-1.amazonaws.com"
```
203.0.113.0/24 is a reserved block that can safely be used in documentation.  The same can be used for public_dns_name below.

File: plugins/modules/ec2.py [hunk 3]
```diff
@@ -577,6 +577,297 @@
...
+            sample: {
+                "capacity_reservation_preference": "open"
+            }
+        client_token:
+            description: The idempotency token you provided when you launched the instance, if applicable.
+            type: str
+            returned: always
+            sample: ""
+        cpu_options:
+            description: The idempotency token you provided when you launched the instance, if applicable.
```
Comment (jillr) [inline plugins/modules/ec2.py:753]:
```suggestion
            description: The CPU options for the instance.
```

**Review by jillr — COMMENTED** [review]
Thanks @alinabuzachis, just a couple small comments!

File: plugins/modules/ec2.py (hunk 1 above)
Comment (alinabuzachis) [reply plugins/modules/ec2.py]: Yes, you're right. In the copy/paste process I forget to rectify. Thank you.

File: plugins/modules/ec2.py (hunk 2 above)
Comment (alinabuzachis) [reply plugins/modules/ec2.py]: @jillr Thank you. Just added it (even for public_dns_name).

File: plugins/modules/ec2.py (hunk 3 above)
Comment (alinabuzachis) [reply plugins/modules/ec2.py]: @jillr Thanks.

**Review by jillr — APPROVED** [review]

**Pull request merged**